nlgeval/pycocoevalcap/meteor/meteor.py

```diff
@@ -3,23 +3,33 @@
 # Python wrapper for METEOR implementation, by Xinlei Chen
 # Acknowledge Michael Denkowski for the generous discussion and help 
 
+import sys
 import os
 import subprocess
 import threading
 
-import six
-
 # Assumes meteor-1.5.jar is in the same directory as meteor.py.  Change as needed.
 METEOR_JAR = 'meteor-1.5.jar'
 
 
+def conditional_enc(s):
+    if sys.version_info[0] == 3 and sys.version_info[1] < 6:
+        return s.encode('utf-8')
+    return s
+
+def conditional_dec(s):
+    if sys.version_info[0] == 3 and sys.version_info[1] < 6:
```

Comment:
These checks could add extra time and since they're running in a class we could cache them in the constructor with something like:
```
def unicode_encode(s):
    return s.encode('utf-8')

def noop_encode(s):
    return s
...
def __init__(self):
    ...
    if sys.version_info[0] == 3 and sys.version_info[1] < 6:
        self.encode = unicode_encode
    else:
        self.encode = noop_encode
```

Reply:
I didn't use lambdas because lambdas aren't nice with threading.